streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -583,6 +627,172 @@ public void advanceWallClockTime(final long advanceMs) {
         return new ProducerRecord<>(record.topic(), record.partition(), record.timestamp(), key, value, record.headers());
     }
 
+
+    private final Queue<ProducerRecord<byte[], byte[]>> getRecordsQueue(final String topicName) {
+        final Queue<ProducerRecord<byte[], byte[]>> outputRecords = outputRecordsByTopic.get(topicName);
+        if (outputRecords == null) {
+            if (!processorTopology.sinkTopics().contains(topicName)) {
```

Comment:
Why do we handle this case differently?

Reply:
To my understanding null can happen either topic does not exist or no input piped to that topic. 
This is to able to Throw error if topic does not exist at all.

Reply:
Ack.